### PR TITLE
Pass folderLink to OCResource component

### PR DESCRIPTION
**Pull request by JanAckermann**
## Description
Change: Pass folderLink to OcResource component

For more flexibility, the folderLink needs to be passed to the OcResource component

## Related Issue

- Fixes <issue_link>

## How Has This Been Tested?

- test environment:
- test case 1:
- test case 2:
- ...

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests

## Checklist:

- [x] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation added/updated

## Open tasks:

- [ ] ...

### CODE REVIEW

**Review by pascalwengerter — CHANGES_REQUESTED** [review]
Please only merge when the corresponding PR in `web` is close to its finalization to not block the ODS in the meantime 🤓

**Review by kulmann — COMMENTED** [review]
OcResource has a `parentFolderLinkPath` and a `folderLink` which previously both relied on the target route. Adding one property for each seems weird. What about passing a callback instead which handles building a router link from a resource path?

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "owncloud-design-system",
-  "version": "11.3.1",
+  "version": "12.0.0-alpha.22",
```
Comment (pascalwengerter) [inline package.json:3]: Sollte nicht mit gemerged werden:)

**Review by pascalwengerter — APPROVED** [review]
Should be good to go

**Pull request merged**